Masonry/View.jsx

```diff
@@ -120,44 +129,74 @@ export default class Masonry extends React.PureComponent {
     window.removeEventListener('resize', this.onResize);
   }
 
-  componentWillReceiveProps(nextProps) {
-    if (nextProps.items.length !== this.props.items.length) {
-      this.layout(nextProps);
+  componentDidUpdate (prevProps, prevState) {
+    if (prevProps.items.length !== this.props.items.length) {
+      this.layout(this.props, false)
     }
   }
 
   onResize = throttle(() => {
-    this.layout(this.props, true);
+    this.layout((this.state.props || this.props), true);
   }, 150, { trailing: true })
 
+  dynamicWidth = () => {
+    const _p = this.state.props || this.props
```

Comment:
This is nice! For the most part I relied on a dual CSS/JS combo and only shared the JS portion here.

Reply:
Thanks! I feel this is easier, just toss the number of columns to the component and let it handle it :)